src/sentry/http.py

```diff
@@ -183,3 +221,144 @@ def safe_urlopen(url, method=None, params=None, data=None, json=None,
 
 def safe_urlread(response):
     return response.content
+
+
+def expose_url(url):
+    if url is None:
+        return u'<unknown>'
+    if url[:5] == 'data:':
+        return u'<data url>'
+    url = truncatechars(url, MAX_URL_LENGTH)
+    if isinstance(url, six.binary_type):
+        url = url.decode('utf-8', 'replace')
+    return url
+
+
+def fetch_file(url, domain_lock_enabled=True, outfile=None,
+               headers=None, allow_redirects=True, verify_ssl=False,
+               timeout=settings.SENTRY_SOURCE_FETCH_SOCKET_TIMEOUT, **kwargs):
+    """
+    Pull down a URL, returning a UrlResult object.
+    """
+    # lock down domains that are problematic
+    if domain_lock_enabled:
+        domain = urlparse(url).netloc
+        domain_key = 'source:blacklist:v2:%s' % (
+            md5_text(domain).hexdigest(),
+        )
+        domain_result = cache.get(domain_key)
+        if domain_result:
+            domain_result['url'] = url
+            raise CannotFetch(domain_result)
+
+    logger.debug('Fetching %r from the internet', url)
+
+    http_session = build_session()
+    response = None
+
+    try:
+        try:
+            start = time.time()
+            response = http_session.get(
+                url,
+                allow_redirects=allow_redirects,
+                verify=verify_ssl,
```

Comment:
That's fair, that's why I didn't have a strong opinion. Also `verify` by itself is a bit ambiguous.